components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/ExceptionCodes.java

```diff
@@ -127,9 +127,19 @@
     TEMPLATE_EXCEPTION(900501, "Service configuration Error", 500, " Error generate service config"),
     GATEWAY_EXCEPTION(900502, "Gateway publishing Error", 500, " Error occurred while publishing to Gateway"),
     BROKER_EXCEPTION(900503, "Broker Connection Error", 500, " Error occurred while obtaining broker connection"),
-    GATEWAY_ENVIRONMENT_NOT_FOUND(900504, "Invalid Gateway Environment", 400, "Gateway Environment with name '%s' not found"),
+    INVALID_GATEWAY_ENVIRONMENT(900504, "Invalid Gateway Environment", 400, "Gateway Environment with name '%s' not found"),
     NO_GATEWAY_ENVIRONMENTS_ADDED(900505, "No Gateway Environments Available", 400, "No gateway environments " +
             "available for the API : %s."),
+    GATEWAY_ENVIRONMENT_NOT_FOUND(900506, "Gateway Environment not found", 404,
+            "Gateway Environment with %s not found"),
+    EXISTING_GATEWAY_ENVIRONMENT_FOUND(900507, "Gateway Environment already exists", 400,
+            "A Gateway Environment with %s already exists"),
+    READONLY_GATEWAY_ENVIRONMENT(900508, "Gateway Environment is read only", 400,
+            "A Gateway Environment with %s is read only"),
+    GATEWAY_ENVIRONMENT_DUPLICATE_VHOST_FOUND(900509, "Gateway Environment with duplicate virtual hosts",
+            400, "A Gateway Environment can not exists with duplicate virtual hosts"),
```

Comment:
Preferred is Cannot over can not

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/9890/commits/27c308ed7cd59dba7a8323262bcd516d7168e9fd